coalahtml/_coalahtml/app/controllers/main.js

```diff
@@ -1,41 +1,32 @@
 'use strict';
 
 angular.module('coalaHtmlApp')
-  .controller('MainCtrl', function ($scope, $http) {
-    var parseCoalaProject = function(projectDir, coalaJSON) {
-      $http.get(projectDir + $scope.Constants.file_data)
-        .success(function(file_data) {
-          $scope.file_dict = angular.copy(file_data);
-          coalaJSON = coalaJSON || projectDir + $scope.Constants.coala;
-          $scope.data = [];
-          $http.get(coalaJSON).success(function(coala_json) {
-            var knownFiles = {};
-            var parseResult = function(el) {
-                el.affected_code.forEach(function(sourceRange){
-                    if (!(sourceRange.file in knownFiles)){
-                        knownFiles[sourceRange.file] = [];
-                    }
-                    knownFiles[sourceRange.file].push({
-                      "start":    sourceRange.start.line,
-                      "end":      sourceRange.end.line,
-                      "diffs":    el.diffs,
-                      "message":  el.message,
-                      "origin":   el.origin,
-                      "severity": el.severity
-                    });
-                });
-            };
-            for (var section in coala_json.results) {
-              if (coala_json.results[section].length > 0) {
-                coala_json.results[section].forEach(parseResult);
-              }
+  .controller('MainCtrl', function ($scope, $rootScope) {
+    var parseCoalaProject = function() {
+        $scope.data = [];
+        var knownFiles = {};
+        var parseResult = function(el) {
+          el.affected_code.forEach(function(sourceRange) {
+            if (!(sourceRange.file in knownFiles)) {
+              knownFiles[sourceRange.file] = [];
             }
-            $scope.data.push(knownFiles);
+            knownFiles[sourceRange.file].push({
+              "start":    sourceRange.start.line,
+              "end":      sourceRange.end.line,
+              "diffs":    el.diffs,
+              "message":  el.message,
+              "origin":   el.origin,
+              "severity": el.severity
+            });
           });
-        });
+        };
+        for (var section in $rootScope.COALA_JSON.results) {
+          if ($rootScope.COALA_JSON.results[section].length > 0) {
+            $rootScope.COALA_JSON.results[section].forEach(parseResult);
+          }
+        }
+        $scope.data.push(knownFiles);
       };
-    $http.get("data/Constants.json").success(function(constants) {
-      $scope.Constants = constants;
-      parseCoalaProject($scope.Constants.data);
-    });
+
+    parseCoalaProject();
```

Comment:
I had similar opinion. But I didn't remove the function because in future
we can easily separate the code that need not belong to the function. So we
might need it?
ᐧ

On Fri, Jun 3, 2016 at 3:28 PM, Attila <notifications@github.com> wrote:

> In coalahtml/_coalahtml/app/controllers/main.js
> <https://github.com/coala-analyzer/coala-html/pull/52#discussion_r65683873>
> :
>
> >        };
> > -    $http.get("data/Constants.json").success(function(constants) {
> > -      $scope.Constants = constants;
> > -      parseCoalaProject($scope.Constants.data);
> > -    });
> > +
> > +    parseCoalaProject();
>
> Why do you need create a function and call it straight away. Is this
> function is used (will be used) anywhere else?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-html/pull/52/files/b7f706d2171c77e341e20b5b680c0c87b1b38b7e#r65683873>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHDgOYXslWGacCM1B5HTjQkbxeFgGw3Qks5qH_qpgaJpZM4IslPq>
> .
>



-- 
Best regards,
Tushar Gautam
http://gtushar.co